### DISCUSSION TITLE
How do know which http transport is in active state?

### DISCUSSION THREAD

**Discussion by kfazil** [wevm/viem#2685] [Question]
```
const client = createPublicClient({
        chain: chain,        
        transport: fallback([
            chainStackHttpTransport(chain),
            infuraHttpTransport(chain),
            oneRPCHttpTransport(chain),
            ankrHttpTransport(chain),
            alchemyHttpTransport(chain),
        ])
    });
```

is there any way we can check which http transport is client going to use while making request?

**Comment by jxom** [comment]
Where exactly do you want to check this?

**Reply by kfazil** [reply]
I've created an API route in Next.js where I'm using this. That route will also log the call count using an external API. So, before making the `client.request` JSON-RPC call, I want to log the HTTP transport name that will be used to make the request.

More specifically, I want to access the HTTP transport from the transport fallback array, which `usePublicClient` will use when a JSON-RPC request is made on the UI.
